tests/unit/test_handler.py

```diff
@@ -277,19 +284,23 @@ def test_parse_extra():
     assert parsed_extra_codes == expected_extra_codes
 
 
+def test_process_chart():
```

Comment:
I don't see a test for the new `filter_chart` function?  or did i miss it?

Reply:
wow, thanks, while adding a test for it I found a bug in the filter parameter processing. more tests incoming